From 9dd434fd6d94d0e80f645bb490522f43ea8add85 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:44 -0400
Subject: [ethernet] igb: fix warning if !CONFIG_IGB_HWMON

Message-id: <1409062359-13063-27-git-send-email-sassmann@redhat.com>
Patchwork-id: 89146
O-Subject: [RHEL7.1 PATCH 26/81] igb: fix warning if !CONFIG_IGB_HWMON
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jeff Kirsher <jeffrey.t.kirsher@intel.com>

Fix warning about code defined but never used if IGB_HWMON not defined.

Reported-by: Stephen Hemminger <stephen@networkplumber.org>
Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 9b143d11a43aa7c188d53a996cdc9172e5b4b4b0)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/e1000_82575.c b/drivers/net/ethernet/intel/igb/e1000_82575.c
index 1da4e87..fa36fe1 100644
--- a/drivers/net/ethernet/intel/igb/e1000_82575.c
+++ b/drivers/net/ethernet/intel/igb/e1000_82575.c
@@ -2711,6 +2711,7 @@ static const u8 e1000_emc_therm_limit[4] = {
  E1000_EMC_DIODE3_THERM_LIMIT
 };
 
+#ifdef CONFIG_IGB_HWMON
 /**
  *  igb_get_thermal_sensor_data_generic - Gathers thermal sensor data
  *  @hw: pointer to hardware structure
@@ -2833,6 +2834,7 @@ static s32 igb_init_thermal_sensor_thresh_generic(struct e1000_hw *hw)
  return status;
 }
 
+#endif
 static struct e1000_mac_operations e1000_mac_ops_82575 = {
  .init_hw              = igb_init_hw_82575,
  .check_for_link       = igb_check_for_link_82575,
-- 
1.7.1